src/3d/symbols/qgspointcloud3dsymbol_p.cpp

```diff
@@ -227,6 +267,179 @@ void QgsPointCloud3DSymbolHandler::makeEntity( Qt3DCore::QEntity *parent, const
   // cppcheck-suppress memleak
 }
 
+void QgsPointCloud3DSymbolHandler::triangulate( QgsPointCloudIndex *pc, const IndexedPointCloudNode &n, const QgsPointCloud3DRenderContext &context, const QgsAABB &bbox )
+{
+  if ( outNormal.positions.isEmpty() )
+    return;
+
+  bool hasColorData = !outNormal.colors.empty();
+  bool hasParameterData = !outNormal.parameter.empty();
+
+  // first, get the points of the concerned node
+  std::vector<double> vertices( outNormal.positions.size() * 2 );
+  size_t idx = 0;
+  for ( int i = 0; i < outNormal.positions.size(); ++i )
+  {
+    vertices[idx++] = outNormal.positions.at( i ).x();
+    vertices[idx++] = outNormal.positions.at( i ).z();
+  }
+
+  // next, we also need all points of all parents nodes to make the triangulation (also external points)
+  IndexedPointCloudNode parentNode = n.parentNode();
+
+  int properPointsCount = outNormal.positions.count();
+  while ( parentNode.d() >= 0 )
+  {
+    processNode( pc, parentNode, context );
+    parentNode = parentNode.parentNode();
+  }
+
+  PointData filteredExtraPointData;
+
+  double span = pc->span();
+  double extraBoxFactor = 16 / span;
```

Comment:
This allows to have extra band out the box with width depending of the density of points. Less point we have, more large is the band to get points further.
I've tried different value, and 16 give me good results. I was thinking of adding a settings to let the user change this value, but I'd prefer to keep like this to avoid too much settings and I find this one not obvious.

Reply:
Looking at untwine code, it currently uses fixed span=128 all the time:
https://github.com/hobu/untwine/blob/main/bu/BuPyramid.cpp#L90

So in fact 16/128 = 12.5% extra buffer (at least for datasets that we index with untwine).

12.5% seems like quite a lot, but if you tried some tuning of it, and this worked best, I am happy with it - but it would be probably useful to add some comments about that calculation and what extraBoxFactor to expect (most readers probably won't know that span is usually 128)